fiftyone/brain/internal/core/pgvector.py

```diff
@@ -0,0 +1,142 @@
+"""
+PGVector similarity backend.
+
+| Copyright 2017-2024, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import logging
+import numpy as np
+import psycopg2
+import eta.core.utils as etau
+import fiftyone.core.utils as fou
+from fiftyone.brain.similarity import (
+    SimilarityConfig,
+    Similarity,
+    SimilarityIndex,
+)
+import fiftyone.brain.internal.core.utils as fbu
+
+pgvector = fou.lazy_import("pgvector")
+logger = logging.getLogger(__name__)
+
+_SUPPORTED_METRICS = ("cosine", "dotproduct", "euclidean")
+
+class PgVectorSimilarityConfig(SimilarityConfig):
```

Comment:
If you look at the elasticsearch integration they allow for defining the distance metric. We should do the same:
https://github.com/voxel51/fiftyone-brain/blob/e3300432b83adcb515a9179220c7f1f6979d1448/fiftyone/brain/internal/core/elasticsearch.py#L62

We will also need the SSL certs and other connection types that people use with PostgreSQL